### PR TITLE
Improve javadocs for querydsl-apt

### PR DESCRIPTION
**Pull request by timowest**
related to https://github.com/querydsl/querydsl/issues/1190

### CODE REVIEW

File: querydsl-apt/src/main/java/com/querydsl/apt/VisitorConfig.java [hunk 1]
```diff
@@ -14,7 +14,7 @@
 package com.querydsl.apt;
 
 /**
- * VisitorConfig defines the entity type specific visiting configuration
+ * {@link VisitorConfig} defines the entity type specific visiting configuration
```
Comment (johnktims) [inline querydsl-apt/src/main/java/com/querydsl/apt/VisitorConfig.java]: This sentence seems a little weird but I couldn't think of a better way to rewrite it.

File: querydsl-apt/src/main/java/com/querydsl/apt/SimpleTypeVisitorAdapter.java [hunk 2]
```diff
@@ -13,19 +13,21 @@
...
 import java.lang.reflect.Method;
 import java.util.List;
 
+import javax.lang.model.type.IntersectionType;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.util.SimpleTypeVisitor6;
+
 /**
- * Converts Java 8 IntersectionType instances into their first bound when visiting
+ * Converts Java 8 {@link IntersectionType} instances into their first bound when visiting
```
Comment (johnktims) [inline querydsl-apt/src/main/java/com/querydsl/apt/SimpleTypeVisitorAdapter.java]: This sentence also seems a little weird but I don't completely understand what it's describing. I'll review it again tomorrow.

Comment (Shredder121) [inline querydsl-apt/src/main/java/com/querydsl/apt/SimpleTypeVisitorAdapter.java]: While visiting the AST, it [flattens the IntersectionType model to just the first bound](https://github.com/querydsl/querydsl/blob/apt-javadocs/querydsl-apt/src/main/java/com/querydsl/apt/SimpleTypeVisitorAdapter.java#L46-L47).
So `Number & Comparable` would just be `Number`.

File: querydsl-apt/src/main/java/com/querydsl/apt/VisitorConfig.java (hunk 1 above)
Comment (Shredder121) [inline querydsl-apt/src/main/java/com/querydsl/apt/VisitorConfig.java]:
```java
/**
* {@link VisitorConfig} defines visiting configuration specific to entity types
*/

```

Maybe throw in an `{@link }` for `EntityType`?

File: querydsl-apt/src/main/java/com/querydsl/apt/VisitorConfig.java [hunk 3]
```diff
@@ -13,8 +13,10 @@
  */
 package com.querydsl.apt;
 
+import com.querydsl.codegen.EntityType;
+
 /**
- * VisitorConfig defines the entity type specific visiting configuration
+ * {@code VisitorConfig} defines the {@link EntityType}-specific visiting configuration
```
Comment (Shredder121) [inline querydsl-apt/src/main/java/com/querydsl/apt/VisitorConfig.java]: `{@linkplain }` reads better if you want it to directly flow in the sentence(like it does here).

Comment (timowest) [inline querydsl-apt/src/main/java/com/querydsl/apt/VisitorConfig.java]: But according to the docs it is mostly useful when the link text is not code. At least the JDK javadocs use `@code` quite a lot for self references.

**Pull request merged**